operator/v1/types_console.go

```diff
@@ -143,8 +143,73 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeType defines the type of the theme for the console UI.
+type ThemeType string
+
+// ThemeType values
+const (
+    // DarkTheme represents the dark theme for the console UI.
+    DarkTheme ThemeType = "Dark"
+    // LightTheme represents the light theme for the console UI.
+    LightTheme ThemeType = "Light"
+)
+
+// LogoType defines the type of the logo for the console UI.
+type LogoType string
+
+const (
+    // Masthead represents the logo in the masthead.
+    Masthead LogoType = "Masthead"
+    // Favicon represents the favicon logo.
+    Favicon LogoType = "Favicon"
+)
+
+// CustomLogos defines a configuration based on theme types for the console UI logo.
+type CustomLogo struct {
+    // type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+    // This field is an enum with valid values "Masthead" and "Favicon".
+    // +kubebuilder:validation:Enum="Masthead";"Favicon"
+    // +required
+    Type LogoType `json:"type"`
+    // theme specifies the theme type for the console UI logo. It determines whether the console should use the dark or light theme.
+    // This field is an enum with valid values "Dark" and "Light".
+    // +kubebuilder:validation:Enum="Dark";"Light"
+    // +required
+    Theme ThemeType `json:"theme"`
+    // path is the reference to a specific file within a ConfigMap in the openshift-config namespace.
+	// This field allows the console to locate and use the specified file containing a custom logo.
+	// The ConfigMap must be created with the appropriate keys and file extensions to ensure the console serves the file with the correct MIME type.
+	// The keys are "Masthead-Dark", "Masthead-Light", "Favicon-Dark", and "Favicon-Light".
```

Comment:
If you are requiring specific keys be used, does this need to be a `configv1.ConfigMapFileReference`? Would having only the name reference of the configmap be sufficient?